store/postgres/src/mapping.rs

```diff
@@ -0,0 +1,1141 @@
+                    tables.push(table);
+                }
+                TypeDefinition(Interface(interface_type)) => {
+                    interfaces.insert(interface_type.name.clone(), vec![]);
+                }
+                other => {
+                    return Err(StoreError::Unknown(format_err!(
+                        "can not handle {:?}",
+                        other
+                    )))
+                }
+            }
+        }
+
+        // Resolve references
+        for table in tables.iter_mut() {
+            for column in table.columns.iter_mut() {
+                let named_type = named_type(&column.field_type);
+                if is_scalar_type(named_type) {
+                    // We only care about references
+                    continue;
+                }
+
+                let references = match interfaces.get(named_type) {
+                    Some(table_names) => {
+                        // sql_type is an interface; add each table that contains
+                        // a type that implements the interface into the references
+                        table_names
+                            .iter()
+                            .map(|table_name| Reference::to_column(table_name.clone(), column))
+                            .collect()
+                    }
+                    None => {
+                        // sql_type is an object; add a single reference to the target
+                        // table and column
+                        let other_table_name = Table::collection_name(named_type);
+                        let reference = Reference::to_column(other_table_name, column);
+                        vec![reference]
+                    }
+                };
+                column.references = references;
+            }
+        }
+
+        let root_table = Mapping::make_root_table(&tables, id_type);
+
+        let tables: Vec<_> = tables.into_iter().map(|table| Rc::new(table)).collect();
+        let interfaces = interfaces
+            .into_iter()
+            .map(|(k, v)| {
+                // The unwrap here is ok because tables only contains entries
+                // for which we know that a table exists
+                let v: Vec<_> = v
+                    .iter()
+                    .map(|name| {
+                        tables
+                            .iter()
+                            .find(|table| &table.name == name)
+                            .unwrap()
+                            .clone()
+                    })
+                    .collect();
+                (k, v)
+            })
+            .collect::<HashMap<_, _>>();
+
+        let count_query = tables
+            .iter()
+            .map(|table| format!("select count(*) from \"{}\".\"{}\"", schema, table.name))
+            .collect::<Vec<_>>()
+            .join("\nunion all\n");
+        let count_query = format!("select sum(e.count) from ({}) e", count_query);
```

Comment:
This count query is no longer correct now that we store historical data in the same table as the current entities.

Reply:
Commit 8ead5400